validator/client/src/main/java/tech/pegasys/teku/validator/client/signer/ExternalSigner.java

```diff
@@ -59,92 +65,127 @@
   private final HttpClient httpClient;
   private final ThrottlingTaskQueue taskQueue;
 
+  private final Counter successCounter;
+  private final Counter failedCounter;
+  private final Counter timeoutCounter;
+
   public ExternalSigner(
       final HttpClient httpClient,
       final URL signingServiceUrl,
       final BLSPublicKey blsPublicKey,
       final Duration timeout,
-      final ThrottlingTaskQueue taskQueue) {
+      final ThrottlingTaskQueue taskQueue,
+      final MetricsSystem metricsSystem) {
     this.httpClient = httpClient;
     this.signingServiceUrl = signingServiceUrl;
     this.blsPublicKey = blsPublicKey;
     this.timeout = timeout;
     this.taskQueue = taskQueue;
+
+    final LabelledMetric<Counter> labelledCounter =
+        metricsSystem.createLabelledCounter(
+            TekuMetricCategory.VALIDATOR,
+            "external_signer_requests",
+            "Completed external signer counts",
+            "success",
+            "failed",
+            "timeout");
+    successCounter = labelledCounter.labels("success");
+    failedCounter = labelledCounter.labels("failed");
+    timeoutCounter = labelledCounter.labels("timeout");
   }
 
   @Override
   public SafeFuture<BLSSignature> createRandaoReveal(final UInt64 epoch, final ForkInfo forkInfo) {
     return sign(
-        signingRootForRandaoReveal(epoch, forkInfo),
-        SignType.RANDAO_REVEAL,
-        Map.of("randao_reveal", Map.of("epoch", epoch), FORK_INFO, forkInfo(forkInfo)),
-        slashableGenericMessage("randao reveal"));
+            signingRootForRandaoReveal(epoch, forkInfo),
+            SignType.RANDAO_REVEAL,
+            Map.of("randao_reveal", Map.of("epoch", epoch), FORK_INFO, forkInfo(forkInfo)),
+            slashableGenericMessage("randao reveal"))
+        .whenComplete(this::recordMetrics);
```

Comment:
Maybe just put this inside the `sign` method so it doesn't need to be duplicated?

This `.whenComplete` would just become the last line in `sign` (ie  `SafeFuture.of(...).whenComplete(this::recordMetrics)`)

Reply:
good call, i was focused on the main change and should have cleaned that up.